Implement approve threshold for license and vote

Ankarrr opened the pull request:
- Implement approve threshold for license and vote 
- Add approve threshold/time to listproperties RPC

Ankarrr committed:
Implement approve threshold for license and vote

- push approveThreshold into tx payload in CreatePayload_IssuanceManaged()
- interpret approve_threshold in interpret_CreatePropertyManaged()
- save approve_threshold for the property in logicMath_CreatePropertyManaged()
- compare approve_threshold and approve_time when Grant tokens. if (approve_time < approve_threshold), grant tokens failed

Ankarrr committed:
Add approve threshold/time to listproperties RPC

Ankarrr commented:
#12  #13

Ankarrr committed:
Set license approve threshold to 2/3*Alliances

- Set license approve threshold to 2/3 of Alliances amount
- Define the threshold percentage in omnicore.h

Ankarrr committed:
Check tx sender in logicMath_VoteForLicense()

Vote can be valid when sender is one of the alliances

m04120310 commented on src/omnicore/tx.cpp:
```diff
@@ -1713,7 +1719,7 @@ int CMPTransaction::logicMath_CreatePropertyManaged()
     newSP.update_block = newSP.creation_block;
-    newSP.approve_threshold = 0;
+    newSP.approve_threshold = approve_threshold;
     newSP.approve_time = 0;
```
我上次沒注意到
approve/reject time要不要改名成count啊?
time會不會有一點容易搞錯

Ankarrr replied:
喔好啊

Ankarrr committed:
Replace 'time' to 'count' for avoiding confusion

replace 'approve_time' to 'approve_count'
replace 'reject_time' to 'reject_count'

Ankarrr commented:
#23

m04120310 commented on src/omnicore/tx.cpp:
```diff
@@ -1967,8 +1979,13 @@ int CMPTransaction::logicMath_VoteForLicense() {
     }
 
+    // compare sender with Alliance
+    if (!allianceInfoDB->hasAllianceInfo(sender)) {
```
看到這個才發現可能會有問題，現在的hasAllianceInfo()只會看有法有這個info，但不代表這個address已經approved。我有發issue，要實做一個isAllianceApproveds()的method

m04120310 replied:
#27
